mlflow/pytorch/__init__.py

```diff
@@ -325,6 +376,34 @@ def save_model(
         f.write(pickle_module.__name__)
     # Save pytorch model
     model_path = os.path.join(model_data_path, _SERIALIZED_TORCH_MODEL_FILE_NAME)
+
+    if requirements_file:
+        if not isinstance(requirements_file, str):
+            raise TypeError("Path to requirements file should be a string")
+
+        with TempDir() as tmp_requirements_dir:
+            saved_requirements_dir_subpath = "requirements"
+            _download_artifact_from_uri(
+                artifact_uri=requirements_file, output_path=tmp_requirements_dir.path()
+            )
+            shutil.move(
+                tmp_requirements_dir.path(), posixpath.join(path, saved_requirements_dir_subpath)
+            )
+
+    if extra_files:
+        if not isinstance(extra_files, list):
+            raise TypeError("Extra files argument should be a list")
+
+        with TempDir() as tmp_extra_files_dir:
+            saved_extra_files_dir_subpath = "artifacts"
+            for extra_file in extra_files:
+                _download_artifact_from_uri(
+                    artifact_uri=extra_file, output_path=tmp_extra_files_dir.path()
+                )
+            shutil.move(
+                tmp_extra_files_dir.path(), posixpath.join(path, saved_extra_files_dir_subpath)
+            )
```

Comment:
I have the same question as https://github.com/mlflow/mlflow/pull/3436/files#r513084383 for `extra_files` (do people log `extra_files` stored remotely in S3 alongside their pytorch models, or are the extra_files usually on the local filesystem)? Happy to defer to your judgment here on what's common for pytorch users